src/components/ConfirmModal.tsx

```diff
@@ -0,0 +1,128 @@
+import React, {ReactNode} from 'react';
+import {StyleProp, TextStyle, ViewStyle} from 'react-native';
+import useWindowDimensions from '@hooks/useWindowDimensions';
+import CONST from '@src/CONST';
+import IconAsset from '@src/types/utils/IconAsset';
+import ConfirmContent from './ConfirmContent';
+import Modal from './Modal';
+
+type ConfirmModalProps = {
+    /** Title of the modal */
+    title?: string;
+
+    /** A callback to call when the form has been submitted */
+    onConfirm: () => void;
+
+    /** A callback to call when the form has been closed */
+    onCancel?: (ref?: React.RefObject<HTMLElement> | undefined) => void;
+
+    /** Modal visibility */
+    isVisible: boolean;
+
+    /** Confirm button text */
+    confirmText?: string;
+
+    /** Cancel button text */
+    cancelText?: string;
+
+    /** Modal content text/element */
+    prompt?: string | ReactNode;
+
+    /** Whether we should use the success button color */
+    success?: boolean;
+
+    /** Is the action destructive */
+    danger?: boolean;
+
+    /** Whether we should disable the confirm button when offline */
+    shouldDisableConfirmButtonWhenOffline?: boolean;
+
+    /** Whether we should show the cancel button */
+    shouldShowCancelButton?: boolean;
+
+    /** Callback method fired when the modal is hidden */
+    onModalHide?: () => void;
+
+    /** Should we announce the Modal visibility changes? */
+    shouldSetModalVisibility?: boolean;
+
+    /** Icon to display above the title */
+    iconSource?: IconAsset;
+
+    /** Styles for title */
+    // eslint-disable-next-line react/forbid-prop-types
+    titleStyles?: StyleProp<TextStyle>;
+
+    /** Styles for prompt */
+    // eslint-disable-next-line react/forbid-prop-types
+    promptStyles?: StyleProp<TextStyle>;
+
+    /** Styles for icon */
+    // eslint-disable-next-line react/forbid-prop-types
+    iconAdditionalStyles?: StyleProp<ViewStyle>;
+
+    /** Whether to center the icon / text content */
+    shouldCenterContent?: boolean;
+
+    /** Whether to stack the buttons */
+    shouldStackButtons?: boolean;
+};
+
+function ConfirmModal({
+    confirmText = '',
+    cancelText = '',
+    prompt = '',
+    success = true,
+    danger = false,
+    onCancel = () => {},
+    shouldDisableConfirmButtonWhenOffline = false,
+    shouldShowCancelButton = true,
+    shouldSetModalVisibility = true,
+    title = '',
+    iconSource,
+    onModalHide = () => {},
+    titleStyles = [],
+    iconAdditionalStyles = [],
+    promptStyles = [],
```

Comment:
No need for styles default values
```suggestion
    titleStyles,
    iconAdditionalStyles,
    promptStyles,
```

Reply:
Right, updated!